pkg/operator/handlers/dapr_handler.go

```diff
@@ -186,6 +186,15 @@ func (h *DaprHandler) ObjectCreated(obj interface{}) {
 
 // ObjectUpdated handles Dapr crd updates
 func (h *DaprHandler) ObjectUpdated(old interface{}, new interface{}) {
+	oldAnnotated := h.isAnnotatedForDapr(old.(*appsv1.Deployment))
```

Comment:
The first if statement, lines 189 - 193, I'm not sure those are needed?

What this does is create a Dapr service if the old deployment doesn't have a Dapr annotation, which isn't a valid use case.

The second if statement is correct.

Reply:
The code should probably be this:

```go
annotated := h.isAnnotatedForDapr(new.(*appsv1.Deployment))
if !annotated {
	h.ObjectDeleted(new)
} else {
        h.ObjectCreated(new)
}
```

Reply:
Actually the logics are equivalent. The current annotation is checked in the Create/Delete method.

It's ok to make it clearer though. 😃

Reply:
And a bit more, we should pass the old deployment to Deleted method.

I think we should move the service add/delete logic out of the ObjectCreated/ObjectDeleted method, what do you think?

Reply:
> I think we should move the service add/delete logic out of the ObjectCreated/ObjectDeleted method, what do you think?

That's the only thing that the handler is in charge of, so I don't see a motivation to move that out. Where did you think to move it to?

Reply:
I mean we should not call the Created/Deleted method in the Updated method directly.

Reply:
let me show you the code :)

Reply:
Re: latest commits, this looks better, thanks.